src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1424,6 +1424,243 @@ public async Task TestRuntimeBaseRouteInNextLinkForPaginatedRestResponse()
             }
         }
 
+        /// <summary>
+        /// For mutation operations, both the respective operation(create/update/delete) + read permissions are needed to receive a valid response.
+        /// In this test, Anonymous role is configured with only create permission.
+        /// So, a create mutation executed in the context of Anonymous role is expected to result in
+        /// 1) Creation of a new item in the database
+        /// 2) An error response containing the error message : "The mutation operation was successful but the current user is unauthorized to view the response due to lack of read permissions"
+        /// 
+        /// A create mutation operation in the context of Anonymous role is executed and the expected error message is validated.
+        /// Authenticated role has read permission configured. A pk query is executed in the context of Authenticated role to validate that a new
+        /// record was created in the database.
+        /// </summary>
+        [TestMethod]
+        [TestCategory(TestCategory.MSSQL)]
+        public async Task ValidateErrorMessageForMutationWithoutReadPermission()
+        {
+            GraphQLRuntimeOptions graphqlOptions = new(Enabled: true);
+            RestRuntimeOptions restRuntimeOptions = new(Enabled: false);
+
```

Comment:
Does this test really belong in ConfigurationTests.cs? Why not in authorization tests, or perhaps more consisely, its own test class. Then you can also add tests for additional edge cases explicitly to what you're testing.

Reply:
I started with a new test class but then realized that a lot of the methods present in the ConfigurationTest class are necessary. This sounds like an opportunity to refactor the helper methods to another common class and update existing tests.  

I'm leaning towards having this test in ConfigurationTests class for now so that it is not blocker for 0.10.* release. There are some more tests that don't necessarily belong in ConfigurationTests class. I can work on a follow-up refactor PR once this gets merged.